### PR TITLE
[travis] Make overfull hboxes a presubmit error.

### PR DESCRIPTION
**Pull request by tkoeppe**
Also delete the "manual" check since "complete" subsumes "manual".

@zygoloid, @godbyk: Could you please take a look whether this looks right?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
-  - BUILD_TYPE=complete  # build manually and regenerate figures, grammer, and cross-references
+  - BUILD_TYPE=complete  # build manually and regenerate figures, grammar, and cross-references
   - BUILD_TYPE=check-whitespace  # check for whitespace at the ends of lines
 
 script:
   # Build std.pdf
   - pushd source
   - if [ "$BUILD_TYPE" = "latexmk" ]; then
       latexmk -pdf std;
+      ! grep -Fe "Overfull \hbox" std.log;
```
Comment (godbyk) [inline .travis.yml]: Need to escape the backslash with another backslash.

**Comment by godbyk** [comment]
The `manual` method was originally to just perform the build steps as specified in the README. But `complete` does subsume `manual` in that it performs the additional step of regenerating the images.

File: .travis.yml (hunk 1 above)
Comment (tkoeppe) [reply .travis.yml]: Done, thanks.

**Comment by tkoeppe** [comment]
OK, thanks, in that case I'd prefer to not rerun redundant steps that we'd catch anyway. No need to waste compute resources and time.

**Pull request merged**